src/query/ast/src/parser/expr.rs

```diff
@@ -1603,12 +1610,11 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
             | #ty_string
             | #ty_variant
             | #ty_nullable
-            ) ~ NOT? ~ NULL? : "type name" },
+            ) ~ #nullable? : "type name" },
         )),
-        |(ty, opt_not, opt_null)| match (opt_not.is_some(), opt_null.is_some()) {
-            (false, true) => Ok(ty.wrap_nullable()),
-            (false, false) => Ok(ty),
-            (true, true) => {
+        |(ty, opt_nullable)| match opt_nullable {
+            Some(true) => Ok(ty.wrap_nullable()),
+            Some(false) => {
                 if matches!(ty, TypeName::Nullable(_)) {
```

Comment:
this is impossible because parser of `ty` contains no nullable

Reply:
We also support `nullable(type)` syntax
for example
```sql
CREATE TABLE t(c1 NULLABLE(int));
```
in this case `c1 NULLABLE(int) NOT NULL` is invalid